src/gui/sharelinkwidget.cpp

```diff
@@ -58,7 +58,14 @@ ShareLinkWidget::ShareLinkWidget(AccountPtr account,
     //Is this a file or folder?
     QFileInfo fi(localPath);
     _isFile = fi.isFile();
-    _ui->nameLineEdit->setText(tr("%1 link").arg(fi.fileName()));
+
+    // Set the default link share name: it cannot be >64 characters
```

Comment:
What is the limit? 64 unicode codepoint? Or 64 utf-8 bytes, or 64 utf-16 character ?

This code assume the count is in utf-16 character.   Even if it is the case, we should still not cut in the middle of a high and low surrogate.